src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java

```diff
@@ -149,12 +152,23 @@ public ServiceTracker serviceTracker(ApplicationEventPublisher publisher, RouteL
 
         @Bean
         OpaQueryProvider opaQueryProvider(ServiceTracker serviceTracker) {
+            // TARGET ARCH: get application-id from request attributes, not from the service-tracker
             return request -> serviceTracker
-                    .findServices(s -> s.hostname().equals(request.getURI().getHost()))
+                    .findServices(s -> {
+                        var appId = s.getMetadata().get("app.contentgrid.com/application-id");
```

Comment:
Having to juggle these strings so deep in here looks less pleasant than how AppService used to expose it... Is it worth it having something else that extends ServiceInstance and gives better accessors, or have some static methods that do this (but all in one place so the label strings are only in one place)?

Reply:
that's only temporary - the PR after the next cleans it up again ;-)

Reply:
maybe I should try and bring that forward into this PR